matrix_messages.rst

```diff
@@ -0,0 +1,92 @@
+Raiden transport messages
+#########################
+
+Definitions
+===========
+* Sender: entity sending the message
+* Recipient: entity able of receiving and decoding the message
+* Message: A data object exchanged between Sender and Recipient
+* Message Field: an attribute of JSON data format that is required to be included in the message
+
+Transport envelope
+==================
+Messages are exchanged in JSON format.
+Message itself contains metadata that allow recipient to verify sender of the message (via use of PK signatures).
+
+Rationale
+---------
+* plaintext JSON messages are easy to read for both humans and machines
+* JSON format is widely used and can be easily validated using JSON-schema (http://json-schema.org/)
+* JSON is easily extensible and extension can be made backward-compatible
+* Message format must not be dependent on the transport used - it should not matter whether the message is sent via Matrix or REST
+* Sender identity verification must not be part of the transport. It should be possible to submit the message over an untrusted channel or delegate.
+
+Example of a message
+-------------------
+``
+{
+    'signature': '0xabcd....01234',
+    'content-type’: 'base-64',
+    'data': 'eyAnaGVhZGVyJzogeyAndHlwZScgOiAndGV4dC1tZXNzYWdlJywgJ3NlbmRlcic6ICAnMHhhYWVlLi4uMTIzNCcsICd0aW1lc3RhbXAnOiAxMjM0NTY3ODkgfSwNCidib2R5JzogeyAnbWVzc2FnZSc6ICdoZWxsbyB3b3JsZCcgfX0='
+}
+``
+
+Envelope fields:
+===============
+* signature - a sender’s signature of the data field
+* content-type - how the data are encoded, i.e.:
```

Comment:
What is the idea behind allowing different content types? IOW why not define one content type?

Reply:
When debugging it's easier to use plaintext, but to save bandwidth (i.e. with IoT) it might be beneficial to have an option for compressed content-type.